packages/fast-usdc/test/exos/advancer.test.ts

```diff
@@ -391,6 +409,57 @@ test('will not advance same txHash:chainId evidence twice', async t => {
   ]);
 });
 
-test.todo(
-  '#10510 zoeTools.localTransfer fails to deposit borrowed USDC to LOA',
-);
+test('returns payment to LP if zoeTools.localTransfer fails', async t => {
+  const {
+    extensions: {
+      services: { advancer },
+      helpers: { inspectLogs, inspectBorrowerFacetCalls, inspectNotifyCalls },
+      mocks: { rejectLocalTransfeferV },
+    },
+    brands: { usdc },
+  } = t.context;
+  const mockEvidence = MockCctpTxEvidences.AGORIC_PLUS_OSMO();
+
+  void advancer.handleTransactionEvent(mockEvidence);
+  rejectLocalTransfeferV();
+
+  await eventLoopIteration();
+
+  t.deepEqual(
+    inspectLogs(0),
+    [
+      '⚠️ deposit to localOrchAccount failed, attempting to return payment to LP',
+      '"[Error: One or more deposits failed: simulated error]"',
+    ],
+    'contract logs report error',
+  );
+
+  const [borrowCall, repayCall] = inspectBorrowerFacetCalls();
+
+  t.is(borrowCall[0], 'borrow');
+  t.like(borrowCall[2], {
+    USDC: {
+      brand: usdc.brand,
+    },
+  });
+
+  t.is(repayCall[0], 'repay', 'repay is called when zt.localTransfer fails');
+  t.is(
+    repayCall[1],
+    borrowCall[1],
+    'same temp borrowSeat is supplied to LP during repay',
+  );
+  t.deepEqual(repayCall[2], borrowCall[2], 'advance amount is returned to LP');
```

Comment:
check `repayCall[2]` against some literal data? maybe the value of an amount?

Reply:
Good feedback, done. Refactored this test a bit as it was confusing. Also added _alerts if `returnToPool` fallback fails_